app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt

```diff
@@ -65,12 +68,23 @@ class AuthenticationViewModel @Inject constructor(
             toastMessage.value = resource.getString(R.string.match_password)
             return
         }
-
+        userRepository.insertUser(
+            User(
+                idState.value.text,
+                nameState.value.text,
+                false,
+                BCrypt.withDefaults().hashToString(
+                    12, passwordState.value.text.toCharArray()
+                )
+            )
+        )
         val registerBody = UserBody(
```

Comment:
Shouldn't this come after a successful registration?
If yes, please move it after the `if(response.data == null)` one line before `switchToLogin`.